crates/core/src/message.rs

```diff
@@ -37,14 +37,14 @@ pub(crate) struct Transaction {
 impl Transaction {
     pub const NULL: &'static Transaction = &Transaction { id: Ulid(0) };
 
-    pub fn new<T: TxType>() -> Self {
+    pub(crate) fn new<T: TxType>() -> Self {
```

Comment:
doesn't matter but since Transaction is pub(crate), by extension this functions are also pub(crate) (and not pub), this is always the case.

This is more explicit though